js/core.js

```diff
@@ -1274,6 +1274,71 @@ var ERMrest = (function (module) {
         };
 
         /**
+         * Formats the presentation value corresponding to this column definition.
+         * @param {String} data The 'formatted' data value.
+         * @param {Object} options The key value pair of possible options with all formatted values in '.values' key
+         * @returns {string} The presenation value.
+         */
+        this.formatPresentation = function(data, options) {
+            
+            var utils = module._formatUtils, values = options.values, keyValues = {}, columns = options.columns;
+
+            /*
+             * TODO: Add code to handle `pre_format` in the annotation
+             */
+
+            if (!this.annotations.contains(module._annotations.COLUMN_DISPLAY) ||
+                !this.annotations.get(module._annotations.COLUMN_DISPLAY).contains("markdown_pattern")) {
+                return data;
+            }
+
+            // Set pattern from the annotation value
+            var pattern = this.annotations.get(module._annotations.COLUMN_DISPLAY).get("markdown_pattern"); // pattern
+
+            // If pattern is not of type string or is undefined/null then return the value as it is
+            if (typeof pattern !== 'string') return data;
+
+            /* 
+             * Code to validate values and set pattern as null if any of the
+             * values turn out to be null or undefined
+             * As well as set key-value pairs in the keyValues object for further usage while templating
+             */
+            for (var i = 0; i < columns.length; i++) {
+                var cname = columns[i].name;
+                
+                keyValues[cname] = values[i];
+
+                // Check for a match for the search string
+                var search = "{{" + cname + "}}";
+
+                // If the search column is found in the pattern
+                if (pattern && pattern.match(search)) {
```

Comment:
This feels wrong to me... I think we should make it so the template engine itself returns null or raises exception if an expansion fails.  Perhaps we can map raw null values into undefined (absent from the template environment) before doing the expansion?

I think it would be OK/preferrable to patch the template engine if necessary to do this, rather than replicating the template expansion logic outside of it like here.  You don't want to have to search for all possible template syntax that can reference/expand a null column.